apps/dc_tools/odc/apps/dc_tools/_stac.py

```diff
@@ -179,10 +179,23 @@ def _find_self_href(item: Document) -> str:
     return self_uri[0]
 
 
+def _get_relative_path(asset_href, self_link):
+    if self_link is None:
+        return asset_href
+
+    self_path = urlparse(self_link).path
+    href_path = urlparse(asset_href).path
```

Comment:
It seems unlikely, but for completeness, I think we should be explicitly checking that both `self_link` and `asset_href` have the same `scheme` and `netloc` (and keeping as an absolute link if they differ).

Reply:
Maybe we could refrain from checking `scheme`, but check for some degree of similarity between the `netloc` values?